src/functions.h

```diff
@@ -166,6 +166,15 @@ inline int32_t lshiftAndSaturateUnknown(int32_t val, uint8_t lshift) {
 	return signed_saturate_operand_unknown(val, 32 - lshift) << lshift;
 }
 
+static constexpr uint32_t charsToIntegerConstant(char a, char b, char c, char d) {
+	return (static_cast<uint32_t>(a)) | (static_cast<uint32_t>(b) << 8) | (static_cast<uint32_t>(c) << 16)
+	       | (static_cast<uint32_t>(d) << 24);
+}
+
+static constexpr uint16_t charsToIntegerConstant(char a, char b) {
+	return (static_cast<uint16_t>(a)) | (static_cast<uint16_t>(a) << 8);
+}
+
```

Comment:
I tested these both using godbolt and by manually inspecting the generated assembly so I'm 85% sure these generate equivalent values to the multichar constants, but someone should definitely look at them with a fresh pair of eyes and thorough consideration of endianness of the target platform.

We could use them to replace a bunch of magic constants through the XML parsing logic, but I'll leave that for someone more ambitious than I in a followup change.

Reply:
These seem like a reasonable alternative approach to me and I trust your testing on them for these cases.